editor.planx.uk/src/@planx/components/Pay/Public/InviteToPayForm.tsx

```diff
@@ -0,0 +1,246 @@
+import React from "react";
+import { useNavigation } from "react-navi";
+import useSWRMutation from "swr/mutation";
+import { PaymentStatus } from "types";
+import ErrorWrapper from "ui/ErrorWrapper";
+import Input from "ui/Input";
+import InputLabel from "ui/InputLabel";
+import ReactMarkdownOrHtml from "ui/ReactMarkdownOrHtml";
+import { object, string } from "yup";
+
+// Passport keys which will be used to display a preview of the session to the payee as part of their journey
+const SESSION_PREVIEW_KEYS = [
+  ["_address", "title"],
+  ["applicant.agent.name.first"],
+  ["applicant.agent.name.last"],
+  ["proposal.projectType"],
+];
+
+export interface InviteToPayFormProps {
+  changePage: () => void;
+  title?: string;
+  description?: string;
+  paymentStatus?: PaymentStatus;
+}
+
+type CreatePaymentRequest = {
+  payeeName: string;
+  payeeEmail: string;
+  applicantName: string;
+  sessionPreviewKeys: Array<KeyPath>;
+};
+
+type FormValues = Omit<CreatePaymentRequest, "sessionPreviewKeys">;
+
+const validationSchema = object({
+  payeeName: string().trim().required("Nominee name required"),
+  payeeEmail: string()
+    .email(
+      "Enter an email address in the correct format, like name@example.com"
+    )
+    .required("Email address required"),
+  applicantName: string().trim().required("Your details required"),
+});
+
+const StyledForm = styled("form")(({ theme }) => ({
+  display: "flex",
+  flexDirection: "column",
+  gap: theme.spacing(3),
+}));
+
+const SubmitButton: React.FC = () => (
+  <Button
+    variant="contained"
+    color="primary"
+    size="large"
+    type="submit"
+    sx={{ alignSelf: "start" }}
+  >
+    {"Email your nominee"}
+  </Button>
+);
+
+const InviteToPayForm: React.FC<InviteToPayFormProps> = ({
+  changePage,
+  title,
+  description,
+  paymentStatus,
+}) => {
+  const sessionId = useStore((state) => state.sessionId);
+  const navigation = useNavigation();
+
+  const postRequest = async (
+    url: string,
+    { arg }: { arg: CreatePaymentRequest }
+  ) => {
+    const response = await fetch(url, {
+      method: "POST",
+      body: JSON.stringify(arg),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (!response.ok) {
+      throw new Error(
+        `Error generating payment request for session ${sessionId}: ${error}`
+      );
+    }
+    return response.json();
+  };
+
+  const { trigger, isMutating, error } = useSWRMutation(
+    `${process.env.REACT_APP_API_URL}/invite-to-pay/${sessionId}`,
+    postRequest
+  );
+
+  const onSubmit = async (values: FormValues) => {
+    const createPaymentRequest: CreatePaymentRequest = {
+      ...values,
+      sessionPreviewKeys: SESSION_PREVIEW_KEYS,
+    };
+    const paymentRequest: PaymentRequest = await trigger(createPaymentRequest);
+    if (!error && paymentRequest.id)
+      redirectToConfirmationPage(paymentRequest.id);
+  };
+
+  // TODO: This fails in the editor
+  // TODO: Test on prod custom domains
```

Comment:
these are good shouts - i guess a lot of custom routes might currently fail in editor but we aren't super explicit about that? since we say this assumes published flows anyways, i think fair to ignore editor failures for now.

Reply:
Yeah, this isn't really a problem atm and isn't solve-able until we tackle the `previewEnvironment` stuff I don't think.